workflow/spikesort.smk

```diff
@@ -5,49 +5,186 @@ from dotenv import load_dotenv
+
+    output:
+        ephys_sync_complete = touch('{sessions}/{task_path}/{session_id}/processed/ephys_sync.done')
+    
+    threads: 32
+
+    priority: 40
+
+    script:
+        "scripts/s04_ephys_sync.py"
+
+rule cell_metrics_processing:
+    input:
+        rec_properties = rec_properties_input,
+        ephys_sync_complete = '{sessions}/{task_path}/{session_id}/processed/ephys_sync.done'
+
+    output:
+        cell_metrics_processing_complete = touch('{sessions}/{task_path}/{session_id}/processed/cell_metrics_processing.done')
+
+    threads: 32
+
+    priority: 50
+
+    script:
+        "scripts/s05_cell_metrics_processing.py"
+
+
+rule cell_metrics_aggregation:
+    input:
+        cell_metrics_processing_complete = '{sessions}/{task_path}/{session_id}/processed/cell_metrics_processing.done'
+
+    output:
+        cell_metrics_aggregation_complete =  touch('{sessions}/{task_path}/{session_id}/processed/cell_metrics_aggregation.done')
+
+    threads: 32
+
+    priority: 60
+
+    script:
+        "scripts/s06_cell_metrics_aggregation.py"
+
+rule cell_metrics_dim_reduction:
+    input:
+        cell_metrics_aggregation_complete = '{sessions}/{task_path}/{session_id}/processed/cell_metrics_aggregation.done'
+
+    output:
+        cell_metrics_dim_reduction_complete =  touch('{sessions}/{task_path}/{session_id}/processed/cell_metrics_dim_reduction.done')
+
+    threads: 32
+
+    priority: 70
+
+    script:
+        "scripts/s07_cell_metrics_dim_reduction.py"
+
+rule cell_metrics_clustering:
+    input:
+        cell_metrics_dim_reduction_complete = '{sessions}/{task_path}/{session_id}/processed/cell_metrics_dim_reduction.done'
+    
+    output:
+        cell_metrics_clustering_complete =  touch('{sessions}/{task_path}/{session_id}/processed/cell_metrics_clustering.done')
+
+    threads: 32
+
+    priority: 80
+
+    script:
+        "scripts/s08_cell_metrics_clustering.py"
+
+
+rule cells_to_xarray:
+    input:
+        ephys_sync_complete = '{sessions}/{task_path}/{session_id}/processed/ephys_sync.done',
+        xr_session = '{sessions}/{task_path}/{session_id}/processed/xr_session.nc',
+        sorting_path = Path('{sessions}/{task_path}/{session_id}/processed/kilosort3/')
+    
+    output:
+        xr_spikes_trials = '{sessions}/{task_path}/{session_id}/processed/xr_spikes_trials.nc',
+        xr_spikes_session = '{sessions}/{task_path}/{session_id}/processed/xr_spikes_session.nc'
+
+    threads: 32
+
+    priority: 85
+
+    script:
+        "scripts/s09_cell_to_xarray.py"
+
+rule session_correlations:
+    input:
+        xr_spikes_session = '{sessions}/{task_path}/{session_id}/processed/xr_spikes_session.nc'
+
+    output:
+        xr_session_correlations = '{sessions}/{task_path}/{session_id}/processed/xr_session_correlations.nc'
+    
+    threads: 32
+
+    priority: 90
+
+    script:
+        "scripts/s10_session_correlations.py"
 
-rule spike_sort_final:
+rule final:
```

Comment:
multiple rules with the same name is not allowed. This will cause conflict to the master Snakefile